src/Microsoft.AspNetCore.SignalR.Core/Internal/SerializedHubMessage.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+
+namespace Microsoft.AspNetCore.SignalR.Internal
+{
+    /// <summary>
+    /// This class is designed to support the framework. The API is subject to breaking changes.
+    /// Represents a serialization cache for a single message.
+    /// </summary>
+    public class SerializedHubMessage
+    {
+        private SerializedMessage _cachedItem1;
+        private SerializedMessage _cachedItem2;
+        private IDictionary<string, byte[]> _cachedItems;
```

Comment:
I think a dictionary is probably overkill. How many protocols are we talking about? Probably be just as quick to find by looping over cached items and matching on name.

Reply:
Yes I gave the same feedback on the commit. There'll never be enough to justify a dictionary. Use a list

Reply:
I think I'd already done that and then a bad rebase wiped it. Will fix.